src/test/workerRunner.ts

```diff
@@ -242,20 +242,23 @@ export class WorkerRunner extends EventEmitter {
       outputDir: baseOutputDir,
       outputPath: (...pathSegments: string[]): string => {
         fs.mkdirSync(baseOutputDir, { recursive: true });
-        return path.join(baseOutputDir, ...pathSegments);
+        return getContainedPath(baseOutputDir, path.join(...pathSegments));
       },
       snapshotPath: (snapshotName: string): string => {
```

Comment:
Sorry for not spotting it earlier, but perhaps we want `...pathSegments: string[]` here, similar to `outputPath`? This way we can make it strictly-backwards-compatible (if we bring back the `sanitizeForFilePath()` call for each path segment), but also allow for the new usecase.

Reply:
This sound good to keep BC. Just to clarify your idea. So `snapshotPath` now takes `pathSegments` like `outputPath` and sanitizes each of them before joining. But `outputPath` will **not** sanitize the path segments which could make the two paths inconsistent. I realize they are independent directories but they mirror each other currently, Is this acceptable?

Also what is your thoughts about the `toMatchSnapshot(name)` api? Would it remain with `name` and then we would just need to do `snapshotPath(name.split(path.sep))`? Or do you have ideas on maintaining this BC as well?

Reply:
Great questions!

I did confuse `snapshotPath()` and `toMatchSnapshot()` apis. The latter one is important, and we cannot make `...pathSegments` there. But we can go with `toMatchSnapshot(name: string | string[], options?)` and let the user pass an array explicitly. That's not as convenient, but seems like a niche feature. Also, passing an array instead of `foo/bar/baz` makes it cross-platform out of the box because you don't need to worry about path separators. WDYT?

Re: different behavior of `outputPath` vs `snapshotPath` - I think that's fine. We can also move sanitization into `toMatchSnapshot` and only apply it when you pass a single string (as before), and avoid sanitization when given an array. This way, `TestInfo` methods will behave similarly, while `toMatchSnapshot` will be backwards compatible. Does this make sense?

Reply:
Ok yeah that clears up my concerns, I think that all makes sense. Here are my changes based on my understanding 93d9632. Let me know if I missed something. Happy to update anything big or small 👍🏼.